storage/storage-interface/src/cached_state_view.rs

```diff
@@ -130,7 +125,7 @@ impl VerifiedStateView {
         StateCache {
             frozen_base: self.speculative_state,
             state_cache: self.state_cache.into_inner(),
-            proofs: self.state_proof_cache.into_inner(),
+            proofs: self.proof_fetcher.get_proof_cache(),
```

Comment:
make it `into_proof_cache()` and avoid cloning

Reply:
Yes, I tried doing that but, the issue is `ProofFetcher` is behind a shared `Arc` pointer and rust won't allow me to move the value out of the shared pointer - Any fancy way you can think of achieving this?

Reply:
use a rwlock as before?

Reply:
I mean, move the RwLock out, wrapping SyncProofFetcher, and the fetcher can just wrap around a hash map.

Reply:
K. I see you need the rwlock inside.. the Arc here is just for dynamic dispatch.. 

I think ideally we can use a box but Box::into_inner is unstable https://doc.rust-lang.org/src/alloc/boxed.rs.html#583-585

So I'd suggest using yet another RwLock here (but always .read() it), then you have the into_inner().

Reply:
Hmm.. changing this to `Arc<RwLock<dyn ProofFetcher>>` also doesn't work - rust is not happy that we are trying to move dynamically sized object out of `RwLock`.  

```
cannot move a value of type dyn proof_fetcher::ProofFetcher: the size of dyn proof_fetcher::ProofFetcher cannot be statically determined E0161
```

I won't bother too much about this as this part will be gone soon with the Async fetching and copy is unavoidable in case of async fetching anyways as we need to aggregate the proofs from multiple threads.

Reply:
yeah, there's no need to wrap the rwlock with a `Arc`?  try that and give up 😬

Reply:
That won't work - I tried it already :D. The issue is dynamically sized objects either needs to be wrapped by `Arc` or `Box` to be stored on stack. Changing this to only `RwLock` has similar error.

```
the trait bound `RwLock<RawRwLock, dyn ProofFetcher>: std::marker::Sized` is not satisfied [E0277] `RwLock<RawRwLock, dyn ProofFetcher>` does not have a constant size known at compile-time
```

Reply:
What if `Prooffetcher` is a box?